src/test/java/BaseTest.java

```diff
@@ -1,11 +1,156 @@
+        viewAllBtn.click();
+
+    }
+
+    public void clickFirstSong()
+    {
+        WebElement firstSong = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//section[@id='songResultsWrapper']//td[@class='title']")));
+        firstSong.click();
+    }
+
+    public void clickAddTo()
+    {
+        WebElement addToBtn = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button[class='btn-add-to']")));
+        addToBtn.click();
+    }
+
+    public void  addToPlayList(String playListName)
+    {
+        //Created Playlist - 'Test1'
+        WebElement playListNameField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//section[@id='songResultsWrapper']//li[contains(text(),'Test1')]")));
+        playListNameField.click();
+    }
+
+    public String getSuccessAlertMessage()
+    {
+        WebElement successAlertMessage = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[@class='success show']")));
+        return successAlertMessage.getText();
+    }
+
+    public void playNextSong()
+    {
+        WebElement playerControlsBtn = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("#mainFooter div[class='side player-controls']")));
+        playerControlsBtn.click();
+        //Thread.sleep(1000);
+        WebElement playNextSongBtn = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("#mainFooter span[class='play']")));
+        playNextSongBtn.click();
+    }
+
+    public boolean isSoundBarVisible()
+    {
+        WebElement soundBar = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img[alt='Sound bars']")));
+        return soundBar.isDisplayed();
+    }
+
+    public void deleteFromPlaylist(String playList) {
+        //Delete Playlist - 'Test1'
+        WebElement playListNameElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//section[@id='playlists']//a[text()='" + playList + "']")));
+        playListNameElement.click();
+        WebElement deletePlaylist= wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button[class='del btn-delete-playlist']")));
+        deletePlaylist.click();
+
+    }
+
+    public void changePlayListName(String playListName, String newPlayListName)
+    {
+        WebElement playListNameElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//section[@id='playlists']//a[text()='" + playListName + "']")));
+        actions.doubleClick(playListNameElement).perform();
+        WebElement playListNameInputField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[name='name']")));
+        playListNameInputField.sendKeys(Keys.chord(Keys.CONTROL,"A",Keys.BACK_SPACE));
+        playListNameInputField.sendKeys(newPlayListName);
+        playListNameInputField.sendKeys(Keys.ENTER);
+
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

**Move these methods to the corresponding class according to the POM.**

**BaseTest** — Implements the necessary methods for work with webdriver.

The BaseTest class contains common setup and teardown methods, as well as any shared functionality or configurations required by multiple test classes.

**BaseTest**

The problem we are trying to solve here is the reuse of common behaviors across test classes avoiding coding duplication and centralizing these actions at one point.

**@BeforeSuite**	- The annotated method will be run before all tests in this suite have run.

**@BeforeMethod** -	The annotated method will be run before all the test methods in the current class have been run.

**@AfterMethod**	- The annotated method will be run after each test method.

_Example without a Base Test Class_

![image](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/c3abad8a-8bc1-4ad0-8a66-23bd19e7af48)